src/main/java/edu/harvard/iq/dataverse/MapLayerMetadataServiceBean.java

```diff
@@ -109,14 +106,14 @@ public MapLayerMetadata findMetadataByDatafile(DataFile datafile){
         First check if the given user has permission to edit this data.
         
     */
-    public boolean deleteMapLayerMetadataObject(MapLayerMetadata mapLayerMetadata, User user){
+    public boolean deleteMapLayerMetadataObject(MapLayerMetadata mapLayerMetadata, DataverseRequest req){
```

Comment:
Should this use `@notnull`? It will create compile time warnings if someone tries to pass null in.

Reply:
The change I made in this method was replacing the user with the request, so that it uses the updated permission check system.

Adding `@notNull` would change the method behavior, so I suggest not adding it now, as it is not in scope, and would require additional testing (Also, the `null` -> `false` semantics might make sense). 

There are other problems in this method (e.g. ignored result on line 121), but, again, not in scope for issue.